app/src/main/java/app/accrescent/client/workers/AutoUpdateWorker.kt

```diff
@@ -9,38 +12,65 @@ import androidx.work.NetworkType
 import androidx.work.PeriodicWorkRequestBuilder
 import androidx.work.WorkManager
 import androidx.work.WorkerParameters
+import app.accrescent.client.Accrescent.Companion.UPDATE_AVAILABLE_CHANNEL
+import app.accrescent.client.R
+import app.accrescent.client.data.PreferencesManager
 import app.accrescent.client.data.RepoDataRepository
+import app.accrescent.client.data.net.AppRepoData
 import app.accrescent.client.util.PackageManager
 import app.accrescent.client.util.getInstalledPackagesCompat
 import dagger.assisted.Assisted
 import dagger.assisted.AssistedInject
+import kotlinx.coroutines.flow.first
 import java.time.Duration
 
 @HiltWorker
 class AutoUpdateWorker @AssistedInject constructor(
     @Assisted val context: Context,
     @Assisted val workerParams: WorkerParameters,
+    private val preferencesManager: PreferencesManager,
     private val repoDataRepository: RepoDataRepository,
     private val packageManager: PackageManager
 ) : CoroutineWorker(context, workerParams) {
     override suspend fun doWork(): Result {
         try {
             repoDataRepository.fetchRepoData()
 
-            context.packageManager.getInstalledPackagesCompat(0)
+            val packagesToUpdate = context.packageManager.getInstalledPackagesCompat(0)
                 .filter { repoDataRepository.appExists(it.packageName) }
                 .filter {
                     repoDataRepository
                         .getAppRepoData(it.packageName)
                         .versionCode > it.longVersionCode
-                }.forEach { packageManager.downloadAndInstall(it.packageName) }
+                }
+
+            if (preferencesManager.automaticUpdates.first()) {
+                packagesToUpdate.forEach { packageManager.downloadAndInstall(it.packageName) }
+            } else {
+                packagesToUpdate.forEachIndexed { index, packageInfo ->
+                    val repoData = repoDataRepository.getAppRepoData(packageInfo.packageName)
+                    showUpdateNotification(index, packageInfo, repoData)
+                }
+            }
         } catch (e: Exception) {
             return Result.failure()
         }
 
         return Result.success()
     }
 
+    private fun showUpdateNotification(index: Int, packageInfo: PackageInfo, repoData: AppRepoData) {
+        val notificationManager = NotificationManagerCompat.from(context)
+
+        val notification = NotificationCompat.Builder(context, UPDATE_AVAILABLE_CHANNEL)
+            .setContentTitle(packageInfo.packageName)
```

Comment:
I think it would be more appropriate to display the application label here instead of the app ID since it's user-facing